syntax/scan_test.go

```diff
@@ -35,6 +35,10 @@ func scan(src interface{}) (tokens string, err error) {
 		case IDENT:
 			buf.WriteString(val.raw)
 		case INT:
+			if val.bigInt != nil {
+				fmt.Fprintf(&buf, "%d", val.bigInt)
+				break
```

Comment:
I'd prefer an "else" to a "break" here, since it makes the symmetry between the two alternatives clearer.